### PR TITLE
New features

### PR DESCRIPTION
**Pull request by RafaelAlfaro**
Some issues were fixed

### CODE REVIEW

File: src/com/jalasoft/search/commons/DigitalUnitConverter.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+     * Calculates Digital Information Unit
+     *
+     * @param sizeFile   : Size to convert
+     * @param unitSource : Initial digital unit (|b|Kb|Mb|Gb|Tb)
+     * @param unitTo     : Final digital unit (|b|Kb|Mb|Gb|Tb)
+     * @return Value calculated
+     */
+    private double calculateDigitalInformationUnit(int sizeFile, String unitSource, String unitTo) {
+        double result = sizeFile;
+        if (digitalInformationUnit.get(unitSource) > digitalInformationUnit.get(unitTo)) {
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/commons/DigitalUnitConverter.java]: Add a blank space before this line. It should be between variable declaration and part of code
Add a comment // to describe what is containing the variable result

File: src/com/jalasoft/search/commons/DigitalUnitConverter.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            result = calculateToLeft(sizeFile, getNumber(unitSource, unitTo));
+        } else if (digitalInformationUnit.get(unitSource) < digitalInformationUnit.get(unitTo)) {
+            result = calculateToRight(sizeFile, getNumber(unitSource, unitTo));
+        }
+
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/commons/DigitalUnitConverter.java]: Remove this space in blank. It should not contain spaces after a curly bracket

File: src/com/jalasoft/search/commons/PathHandler.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ *   it only in accordance with the terms of the license agreement you entered into with Jalasoft.
+ */
+
+package com.jalasoft.search.commons;
+
+import java.nio.file.InvalidPathException;
+import java.nio.file.Paths;
+
+/**
+ * This class will be handling the path
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/commons/PathHandler.java]: It should describe in third person and present time. i.e This class handles the path

File: src/com/jalasoft/search/commons/PathHandler.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+ *
+ * @author rafael
+ * @version 1.0
+ */
+public class PathHandler {
+
+    private String path;
+
+    /**
+     * Constructor: if the path is empty. The path is set with current path
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/commons/PathHandler.java]: If the path is empty then the path is set with current path

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+    private String advanceSearch;
+    private String contains;
+    private boolean inTitle;
+    private boolean insideFile;
+    private String ownerFile;
+    private boolean fileCreated;
+    private boolean fileModified;
+    private boolean fileAccessed;
+    //    private Date initDate;
+//    private Date finalDate;
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java]: Remove these lines if these are not required in the code

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+        advanceSearch = "";
+        contains = "";
+        inTitle = true;
+        insideFile = false;
+        ownerFile = "";
+        fileCreated = false;
+        fileModified = false;
+        fileAccessed = false;
+        //initDate;
+        //finalDate;
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java]: Remove these lines if these are not required in the code

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+//
+//    public Date getFinalDate() {
+//        return finalDate;
+//    }
+//
+//    public SearchCriteria setFinalDate(Date finalDate) {
+//        this.finalDate = finalDate;
+//        return this;
+//    }
+
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java]: Remove these líneas if these are not required in the code. At least these be blocked and waiting some implementation, then comment it

File: test/com/jalasoft/search/commons/TestPathHandler.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    @Test
+    public void isNotValidPathWithColon() {
+        String stringPath = "c:/te:t";
+        PathHandler path = new PathHandler(stringPath);
+        assertFalse(path.isValidPath());
+    }
+
+    @Test
+    public void isNotValidPathWithQuestionMark() {
+        PathHandler path = new PathHandler("c:/te?t");
```
Comment (sofiacespedes) [inline test/com/jalasoft/search/commons/TestPathHandler.java:50]: Is this test restricted to ? and : symbols?

File: test/com/jalasoft/search/commons/TestPathHandler.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+ */
+package com.jalasoft.search.commons;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * <Class description goes here>
```
Comment (sofiacespedes) [inline test/com/jalasoft/search/commons/TestPathHandler.java]: Set the class description here

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+    /**
+     * Returns the attribute value fileCreated
+     *
+     * @return String
+     */
+    public boolean getFileCreated() {
+        return fileCreated;
+    }
+
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java:224]: Missing javadoc for the method below

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+    /**
+     * Returns the attribute value insideFile
+     *
+     * @return boolean
+     */
+    public boolean getInsideFile() {
+        return insideFile;
+    }
+
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java:186]: Missing javadoc for the method below

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 12]
```diff
@@ -0,0 +1,321 @@
...
+    /**
+     * Returns the attribute value ownerFile
+     *
+     * @return String
+     */
+    public String getOwnerFile() {
+        return ownerFile;
+    }
+
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java:205]: Missing java doc for the method below

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 13]
```diff
@@ -0,0 +1,321 @@
...
+    /**
+     * Returns the attribute value contains
+     *
+     * @return String
+     */
+    public String getContains() {
+        return contains;
+    }
+
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java:148]: Missing javadoc for the method below

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 14]
```diff
@@ -0,0 +1,321 @@
...
+        searchPath = "";
+        fileName = "";
+        advanceSearch = "";
+        contains = "";
+        inTitle = true;
+        insideFile = false;
+        ownerFile = "";
+        fileCreated = false;
+        fileModified = false;
+        fileAccessed = false;
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java]: Group variables by type . i.e. boolean and Strings

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 15]
```diff
@@ -0,0 +1,321 @@
...
+    private String searchPath;
+    private String fileName;
+    private String advanceSearch;
+    private String contains;
+    private boolean inTitle;
+    private boolean insideFile;
+    private String ownerFile;
+    private boolean fileCreated;
+    private boolean fileModified;
+    private boolean fileAccessed;
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java]: group variables by type and take into account the java conventions: i.e.
String   driver, searchPath, fileName,
             advancedSearch, contains;

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 16]
```diff
@@ -0,0 +1,321 @@
...
+    private boolean directory;
+    private char sizeCriteria;
+    private String sizeFile;
+    private String sizeType;
+
+    /**
+     * Constructor : The constructor initializes all the attributes
+     */
+    SearchCriteria() {
+
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java]: Remove this line in blank

File: src/com/jalasoft/search/controller/SearchCriteria.java [hunk 17]
```diff
@@ -0,0 +1,321 @@
...
+    /**
+     * Returns the attribute value sizeFile
+     *
+     * @return char
+     */
+    public String getSizeFile() {
+        return sizeFile;
+    }
+
```
Comment (sofiacespedes) [inline src/com/jalasoft/search/controller/SearchCriteria.java:338]: Add javadoc for this method

File: src/com/jalasoft/search/commons/DigitalUnitConverter.java (hunk 1 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/commons/DigitalUnitConverter.java]: The blank space was added and The comment was added that says:
//The "result" variable contains the conversion calculated

File: src/com/jalasoft/search/commons/DigitalUnitConverter.java (hunk 2 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/commons/DigitalUnitConverter.java]: The blank space was removed

File: src/com/jalasoft/search/commons/PathHandler.java (hunk 3 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/commons/PathHandler.java]: The documentation was updated to: "This class handles the path"
Thanks

File: src/com/jalasoft/search/commons/PathHandler.java (hunk 4 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/commons/PathHandler.java]: Updated to: "If the path is empty then the path is set with the current path"
Thanks

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 5 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java]: The lines 31 and 32 were removed.

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 6 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java]: The lines 55 and 56 were removed.

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 7 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java]: the block commented was removed

File: test/com/jalasoft/search/commons/TestPathHandler.java (hunk 8 above)
Comment (RafaelAlfaro) [reply test/com/jalasoft/search/commons/TestPathHandler.java:50]: No. Only for a question mark. the character ":" is allowed in this case.

File: test/com/jalasoft/search/commons/TestPathHandler.java (hunk 9 above)
Comment (RafaelAlfaro) [reply test/com/jalasoft/search/commons/TestPathHandler.java]: Updated to : 
This Class contains the tests for the methods in PathHandler.

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 10 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java:224]: javadoc was added

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 11 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java:186]: javadoc was added

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 12 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java:205]: javadoc was added

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 13 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java:148]: javadoc was added

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 14 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java]: Updated to:
        driver = "";
        searchPath = "";
        fileName = "";
        advanceSearch = "";
        contains = "";
        sizeCriteria = ' ';
        sizeFile = "";
        sizeType = "";
        ownerFile = "";
        inTitle = true;
        insideFile = false;
        fileCreated = false;
        fileModified = false;
        fileAccessed = false;
        fileHidden = false;
        directory = false;

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 16 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java]: the blank space was removed.

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 17 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java:338]: javadoc was added

File: src/com/jalasoft/search/controller/SearchCriteria.java (hunk 15 above)
Comment (RafaelAlfaro) [reply src/com/jalasoft/search/controller/SearchCriteria.java]: Updated to:
    // Attribute used on basic Search
    private String driver;
    private String searchPath;
    private String fileName;
    // Attribute used on advanced search
    private String advanceSearch;
    private String contains;
    private String ownerFile;
    private String sizeFile;
    private String sizeType;
    private char sizeCriteria;
    private boolean inTitle;
    private boolean insideFile;
    private boolean fileCreated;
    private boolean fileModified;
    private boolean fileAccessed;
    private boolean fileHidden;
    private boolean directory;

**Pull request merged**